src/program.cpp

```diff
@@ -1787,6 +1826,34 @@ bool cvk_entry_point::
     return true;
 }
 
+bool cvk_entry_point::build_descriptor_sets_layout_bindings_for_printf_buffer(
+    binding_stat_map& smap) {
+    std::vector<VkDescriptorSetLayoutBinding> layoutBindings;
+    if (m_program->printf_buffer_info().size > 0 &&
+        (m_program->kernel_flags(m_name) &
+         NonSemanticClspvReflectionMayUsePrintf)) {
```

Comment:
`cvk_kernel::uses_printf` should probably become a member of `cvk_entry_point`.

Reply:
Done